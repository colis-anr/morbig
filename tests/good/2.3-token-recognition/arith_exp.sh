echo  $((1 * $((2*3)) * 42))

echo $((x+=`echo 2`))