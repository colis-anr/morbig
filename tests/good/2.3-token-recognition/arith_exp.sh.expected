[
  "Program_LineBreak_CompleteCommands_LineBreak",
  [ "LineBreak_Empty" ],
  [
    "CompleteCommands_CompleteCommands_NewlineList_CompleteCommand",
    [
      "CompleteCommands_CompleteCommand",
      [
        "CompleteCommand_CList",
        [
          "CList_AndOr",
          [
            "AndOr_Pipeline",
            [
              "Pipeline_PipeSequence",
              [
                "PipeSequence_Command",
                [
                  "Command_SimpleCommand",
                  [
                    "SimpleCommand_CmdName_CmdSuffix",
                    [
                      "CmdName_Word",
                      [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                    ],
                    [
                      "CmdSuffix_Word",
                      [
                        "Word",
                        "$((1 * $((2*3)) * 42))",
                        [
                          [
                            "WordArith",
                            [
                              "Word",
                              "1 * $((2*3)) * 42",
                              [
                                [ "WordLiteral", "1 * " ],
                                [
                                  "WordArith",
                                  [
                                    "Word",
                                    "2*3",
                                    [ [ "WordLiteral", "2*3" ] ]
                                  ]
                                ],
                                [ "WordLiteral", " * 42" ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ],
    [ "NewLineList_NewLine" ],
    [
      "CompleteCommand_CList",
      [
        "CList_AndOr",
        [
          "AndOr_Pipeline",
          [
            "Pipeline_PipeSequence",
            [
              "PipeSequence_Command",
              [
                "Command_SimpleCommand",
                [
                  "SimpleCommand_CmdName_CmdSuffix",
                  [
                    "CmdName_Word",
                    [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                  ],
                  [
                    "CmdSuffix_Word",
                    [
                      "Word",
                      "$((x+=`echo 2`))",
                      [
                        [
                          "WordArith",
                          [
                            "Word",
                            "x+=`echo 2`",
                            [
                              [ "WordLiteral", "x+=" ],
                              [
                                "WordSubshell",
                                [ "SubShellKindBackQuote" ],
                                [
                                  "Program_LineBreak_CompleteCommands_LineBreak",
                                  [ "LineBreak_Empty" ],
                                  [
                                    "CompleteCommands_CompleteCommand",
                                    [
                                      "CompleteCommand_CList",
                                      [
                                        "CList_AndOr",
                                        [
                                          "AndOr_Pipeline",
                                          [
                                            "Pipeline_PipeSequence",
                                            [
                                              "PipeSequence_Command",
                                              [
                                                "Command_SimpleCommand",
                                                [
                                                  "SimpleCommand_CmdName_CmdSuffix",
                                                  [
                                                    "CmdName_Word",
                                                    [
                                                      "Word",
                                                      "echo",
                                                      [
                                                        [
                                                          "WordName", "echo"
                                                        ]
                                                      ]
                                                    ]
                                                  ],
                                                  [
                                                    "CmdSuffix_Word",
                                                    [
                                                      "Word",
                                                      "2",
                                                      [
                                                        [
                                                          "WordLiteral", "2"
                                                        ]
                                                      ]
                                                    ]
                                                  ]
                                                ]
                                              ]
                                            ]
                                          ]
                                        ]
                                      ]
                                    ]
                                  ],
                                  [ "LineBreak_Empty" ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [ "LineBreak_Empty" ]
]